### ISSUE TITLE
High Block I/O when looking at docker stats

### ISSUE THREAD

**Issue by safusu** [tidusjar/Ombi#2444]
#### Ombi build Version:

V 3.0.3477

#### Update Branch:

Open Beta

#### Media Sever:

Plex

#### Media Server Version:

1.13.5.5291-6fa5e50a8

#### Operating System:

Ubuntu

#### Ombi Applicable Logs (from `/logs/` directory or the Admin page):

```
2018-08-14 21:08:48.043 -05:00 [Information] 1 scheduled job(s) enqueued.
2018-08-14 21:13:13.682 -05:00 [Information] 1 scheduled job(s) enqueued.
2018-08-14 21:22:58.949 -05:00 [Information] 1 scheduled job(s) enqueued.
2018-08-14 22:57:59.756 -05:00 [Error] StatusCode: Unauthorized, Reason: Unauthorized, RequestUri: https://plex.tv/users/sign_in.json
2018-08-14 22:58:05.555 -05:00 [Error] StatusCode: Unauthorized, Reason: Unauthorized, RequestUri: https://plex.tv/users/sign_in.json
2018-08-14 22:58:19.265 -05:00 [Error] StatusCode: Unauthorized, Reason: Unauthorized, RequestUri: https://plex.tv/users/sign_in.json
2018-08-14 22:58:36.405 -05:00 [Error] StatusCode: Unauthorized, Reason: Unauthorized, RequestUri: https://plex.tv/users/sign_in.json
2018-08-14 22:59:16.853 -05:00 [Error] StatusCode: 429, Reason: , RequestUri: http://api.themoviedb.org/3/movie/314242?api_key=b8eabaf5608b88d0298aa189dd90bf00
2018-08-14 22:59:16.908 -05:00 [Error] StatusCode: 429, Reason: , RequestUri: http://api.themoviedb.org/3/movie/74522?api_key=b8eabaf5608b88d0298aa189dd90bf00
2018-08-14 22:59:16.930 -05:00 [Error] StatusCode: 429, Reason: , RequestUri: http://api.themoviedb.org/3/movie/348240?api_key=b8eabaf5608b88d0298aa189dd90bf00
2018-08-14 22:59:17.115 -05:00 [Error] Retrying RequestUri: http://api.themoviedb.org/3/movie/314242?api_key=b8eabaf5608b88d0298aa189dd90bf00 Because we got Status Code: 429
2018-08-14 22:59:17.115 -05:00 [Error] Retrying RequestUri: http://api.themoviedb.org/3/movie/348240?api_key=b8eabaf5608b88d0298aa189dd90bf00 Because we got Status Code: 429
2018-08-14 22:59:17.115 -05:00 [Error] Retrying RequestUri: http://api.themoviedb.org/3/movie/74522?api_key=b8eabaf5608b88d0298aa189dd90bf00 Because we got Status Code: 429

```

#### Problem Description:

Running linuxserver/ombi container in docker results in large amounts of block I/O. After 2 days, `docker stats` reported 4.45 GB of data written to the disk (compared to 216 MB read). Logs are not helpful for determining source of I/O.

#### Reproduction Steps:

Install most recent docker container, set up Plex, Sonarr, and Radarr connections. 

#### Other

Logs for the day prior (Above is the log file for today) are as follows:

```
2018-08-13 15:19:06.288 -05:00 [Information] 1 scheduled job(s) enqueued.
2018-08-13 19:22:29.899 -05:00 [Information] 1 servers were removed due to timeout
2018-08-13 21:13:51.253 -05:00 [Information] 1 scheduled job(s) enqueued.
2018-08-13 21:25:40.027 -05:00 [Information] 1 scheduled job(s) enqueued.
2018-08-13 21:33:18.215 -05:00 [Information] 1 scheduled job(s) enqueued.
```

Ombi is connected to two servers, one with ~600 movies/~3000 TV episodes and the other with ~1000 movies/~4000 TV episodes. I can only guess that it is the result of the sync feature, but can't confirm. If there is anything else I can provide, please let me know.

**Comment by HowardStark** [comment]
Experiencing the same issue. Would love to know if there's anything I can do. Thanks!

**Comment by strike84** [comment]
I think this is the same issue as I'm having: https://github.com/tidusjar/Ombi/issues/2399

I haven't had the time to try the suggested fix yet since I can't find where sqlite is installed in the container. Maybe I'll just have to install it on my windows/linux mint VM and run it on the db from there.

Still, I don't think this normal behavior. Radarr/Sonarr kinda behave the same way but not NEARLY as much as the ombi container does. I've stopped using it at all until I have the time to try the suggested fix.

If anyone tries the fix please report back if solved it or not.

**Comment by tidusjar** [comment]
Please can you try the suggested fix in the above linked Issue.

`Still, I don't think this normal behavior` To answer this, SQLite writing to a journal  is common, it is managing the state of the transactions so it can commit them or roll them back.

If it is the case that the schedules.db is the cause (from Hangfire scheduling library) then I can add in the option to move that file somewhere else (similar to the `--storage` command). 

If we can find this out it would help a lot.

**Comment by strike84** [comment]
`SQLite writing to a journal is common` This I understand. I was simply saying it's not normal behavior (I think anyway) to do it that frequently. It literally does it every couple of seconds, all day, every day, it just never ends. I've monitored radarr and a couple other containers too which I can see it's writing to a journal but they never do it in such an extent which ombi does.

It would help to know where sqlite is installed so I can just bash into the container and run the command. If i Just try `sqlite` or `sqlite3` it's not found. Maybe it's not installed? I just tried to run `sqlite3` in radarr and that worked. Is sqlite something linuxserver must add to their container or is it included in ombi?

I was thinking of just copying the db to my vm and just run the sqlite command on the db there and copying it back. But I don't know if that'll work. Would be nice to be able to do it in the container.

If it's linuxserver that needs to install sqlite then I'll head over there and ask them.

**Comment by tidusjar** [comment]
Yeah you'd need to ask the lsio team about that.

Regarding radarr/sonarr not doing it, they are not using Hangfire (which
I'm assuming is causing this)

**Comment by strike84** [comment]
I believe this issue is solved now (I think mine is anyway). I tried the fix in the link above and it looks like it worked. But since the schedules.db is stored in appdata and not in the container I had to run the sqlite command on the host.  

When I use my file activity plugin I still see the journal being created and deleted, but it looks like the massive write to the disk has stopped.  I tested this by running all my containers except ombi for 24h and took a note of how high the write count was on the disk, then I did the same but with ombi running and now there was no significant difference.  

I don't know why my file activity plugin says it's still creating and deleting the journal, which it shouldn't with the TRUNCATE option or the OFF option. With TRUNCATE it should just zero the journal and not delete it and with OFF It shouldn't create a journal at all if I understand it correctly.  But hey, as long as the massive writes have stopped me and my disk are happy. :)
